testsuite/tests/letrec-disallowed/float_block_disallowed.ml

```diff
@@ -1,18 +1,35 @@
 (* TEST
    * flat-float-array
-   ** toplevel
+   ** expect
 *)
 
 (* when the -flat-float-array optimization is active (standard in
    OCaml versions up to at least 4.07), creating an array inspects its
    first element to decide whether it is a float or not; it would thus
-   be unsound to allow to recursively define a float value and an
+   be unsound to allow to recursively define avalue and an
```

Comment:
There's a typo here: `avalue`.

The comment was perhaps too general before, but now it's not general enough.  Recursively defining a value and an array starting with that element is fine if the value is known to be a non-float.  More concretely:

```ocaml
(* Fine: z is known to be a non-float *)
let f (z: int) = let rec x = [| y; z |] and y = z in x

(* Disallowed: z is known to be a float, and will be unboxed during array construction *)
let f (z: float) = let rec x = [| y; z |] and y = z in x

(* Disallowed: z is not known to be a non-float, and will be inspected during array construction *)
let f z = let rec x = [| y; z |] and y = z in x
```